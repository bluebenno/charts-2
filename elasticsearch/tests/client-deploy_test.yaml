suite: test client deployment
templates:
  - client-deploy.yaml
  - common-cm.yaml
tests:
  - it: has defaults
    asserts:
      - hasDocuments: { count: 1 }
      - isAPIVersion: { of: apps/v1 }
      - isKind: { of: Deployment }
      - matchSnapshot: { path: spec }
      - equal:
          path: spec.replicas
          value: 1
      - equal:
          path: spec.revisionHistoryLimit
          value: 5
      - equal:
          path: spec.strategy
          value:
            type: RollingUpdate
            rollingUpdate:
              maxSurge: 1
              maxUnavailable: 0
      - equal:
          path: spec.selector
          value:
            matchLabels:
              app.kubernetes.io/name: elasticsearch
              app.kubernetes.io/instance: RELEASE-NAME
              app.kubernetes.io/component: client
      - equal:
          path: spec.template.metadata
          value:
            labels:
              app.kubernetes.io/name: elasticsearch
              app.kubernetes.io/instance: RELEASE-NAME
              app.kubernetes.io/component: client
      - equal:
          path: spec.template.spec.serviceAccountName
          value: RELEASE-NAME-elasticsearch
      - equal:
          path: spec.template.spec.securityContext
          value:
            fsGroup: 1000
      - isNull:
          path: spec.template.spec.initContainers
      - equal:
          path: spec.template.spec.containers[0].name
          value: elasticsearch
      - matchRegex:
          path: spec.template.spec.containers[0].image
          value: (?:[a-z]+/)?([a-z]+)(?::[0-9]+)?
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: Always
      - equal:
          path: spec.template.spec.containers[0].securityContext
          value:
            capabilities:
              add:
                - IPC_LOCK
                - SYS_RESOURCE
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: ES_NODE_NAME
            value: ${HOSTNAME}
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: ES_DISCOVERY_ZEN_PING_UNICAST_HOSTS
            valueFrom:
              configMapKeyRef:
                key: es.discovery.zen.ping.unicast.hosts
                name: RELEASE-NAME-elasticsearch
      - contains:
          path: spec.template.spec.containers[0].ports
          content:
            name: http
            containerPort: 9200
            protocol: TCP
      - contains:
          path: spec.template.spec.containers[0].ports
          content:
            name: transport
            containerPort: 9300
            protocol: TCP
      - equal:
          path: spec.template.spec.containers[0].lifecycle
          value:
            preStop:
              exec:
                command: ["/bin/kill", "-15", "1"]
      - equal:
          path: spec.template.spec.containers[0].readinessProbe
          value:
            httpGet:
              path: /_cluster/health
              port: http
      - equal:
          path: spec.template.spec.containers[0].livenessProbe
          value:
            httpGet:
              path: /_cluster/health?local=true
              port: http
            initialDelaySeconds: 300
      - contains:
          path: spec.template.spec.containers[0].volumeMounts
          content:
            name: settings
            mountPath: /usr/share/elasticsearch/config/elasticsearch_custom.yml
            subPath: elasticsearch_custom.yml
            readOnly: true
      - equal:
          path: spec.template.spec.containers[0].resources
          value:
            limits:
              memory: 2Gi
      - contains:
          path: spec.template.spec.volumes
          content:
            name: settings
            configMap:
              name: RELEASE-NAME-elasticsearch
              items:
                - key: es.settings
                  path: elasticsearch_custom.yml
      - isEmpty:
          path: spec.template.spec.nodeSelector
      - isEmpty:
          path: spec.template.spec.affinity
      - isEmpty:
          path: spec.template.spec.tolerations

  - it: can set replicas
    set:
      client.replicaCount: 2
    asserts:
      - equal:
          path: spec.replicas
          value: 2

  - it: can set revision history limit
    set:
      revisionHistoryLimit: 1
    asserts:
      - equal:
          path: spec.revisionHistoryLimit
          value: 1

  - it: can set strategy type
    set:
      client.strategyType: Recreate
    asserts:
      - equal:
          path: spec.strategy.type
          value: Recreate
      - isNull:
          path: spec.strategy.rollingUpdate

  - it: can set priority class name
    set:
      priorityClassName: high
    asserts:
      - equal:
          path: spec.template.spec.priorityClassName
          value: high

  - it: can configure sysctl init container
    set:
      virtualMemory.enabled: true
    asserts:
      - contains:
          path: spec.template.spec.initContainers
          content:
            name: sysctl
            image: busybox
            imagePullPolicy: Always
            command:
              - sysctl
              - -w
              - vm.max_map_count=262144
            securityContext:
              privileged: true

  - it: can configure virtual memory max map count
    set:
      virtualMemory.enabled: true
      virtualMemory.maxMapCount: 524288
    asserts:
      - equal:
          path: spec.template.spec.initContainers[0].command
          value:
            - sysctl
            - -w
            - vm.max_map_count=524288

  - it: can set image repository and tag
    set:
      image.repository: example
      image.tag: 1
    asserts:
      - equal:
          path: spec.template.spec.containers[0].image
          value: example:1

  - it: can set image pull policy
    set:
      image.pullPolicy: Never
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: Never

  - it: can set an extra env option
    set:
      common.env:
        test.env.option: test
    asserts:
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: TEST_ENV_OPTION
            valueFrom:
              configMapKeyRef:
                name: RELEASE-NAME-elasticsearch
                key: test.env.option

  - it: can set resources
    set:
      client:
        resources:
          limits:
            cpu: 1
            memory: 1Gi
          requests:
            cpu: 100m
            memory: 128Mi
    asserts:
      - equal:
          path: spec.template.spec.containers[0].resources
          value:
            limits:
              cpu: 1
              memory: 1Gi
            requests:
              cpu: 100m
              memory: 128Mi

  - it: can set node selector
    set:
      client:
        nodeSelector:
          failure-domain.beta.kubernetes.io/region: region
    asserts:
      - equal:
          path: spec.template.spec.nodeSelector
          value:
            failure-domain.beta.kubernetes.io/region: region

  - it: can set affinity
    set:
      client:
        affinity:
          nodeAffinity:
            requiredDuringSchedulingIgnoredDuringExecution:
              nodeSelectorTerms:
                - matchExpressions:
                    - key: kubernetes.io/hostname
                      operator: In
                      values:
                        - test
    asserts:
      - equal:
          path: spec.template.spec.affinity
          value:
            nodeAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                nodeSelectorTerms:
                  - matchExpressions:
                      - key: kubernetes.io/hostname
                        operator: In
                        values:
                          - test

  - it: can set tolerations
    set:
      client:
        tolerations:
          - key: key
            operator: Exists
            effect: NoSchedule
    asserts:
      - contains:
          path: spec.template.spec.tolerations
          content:
            key: key
            operator: Exists
            effect: NoSchedule
