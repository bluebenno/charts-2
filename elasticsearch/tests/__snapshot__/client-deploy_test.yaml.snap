has defaults:
  1: |
    replicas: 1
    revisionHistoryLimit: 5
    selector:
      matchLabels:
        app.kubernetes.io/component: client
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: elasticsearch
    strategy:
      rollingUpdate:
        maxSurge: 1
        maxUnavailable: 0
      type: RollingUpdate
    template:
      metadata:
        labels:
          app.kubernetes.io/component: client
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: elasticsearch
      spec:
        containers:
        - env:
          - name: ES_NODE_NAME
            value: ${HOSTNAME}
          - name: ES_DISCOVERY_ZEN_PING_UNICAST_HOSTS
            valueFrom:
              configMapKeyRef:
                key: es.discovery.zen.ping.unicast.hosts
                name: RELEASE-NAME-elasticsearch
          - name: ES_DISCOVERY_ZEN_MINIMUM_MASTER_NODES
            valueFrom:
              configMapKeyRef:
                key: es.discovery.zen.minimum.master.nodes
                name: RELEASE-NAME-elasticsearch
          - name: ES_GATEWAY_RECOVER_AFTER_DATA_NODES
            valueFrom:
              configMapKeyRef:
                key: es.gateway.recover_after_data_nodes
                name: RELEASE-NAME-elasticsearch
          - name: ES_GATEWAY_EXPECTED_DATA_NODES
            valueFrom:
              configMapKeyRef:
                key: es.gateway.expected_data_nodes
                name: RELEASE-NAME-elasticsearch
          - name: ES_GATEWAY_RECOVER_AFTER_TIME
            valueFrom:
              configMapKeyRef:
                key: es.gateway.recover_after_time
                name: RELEASE-NAME-elasticsearch
          image: mikelorant/elasticsearch:v1.0.1-elasticsearch6.5.1
          imagePullPolicy: Always
          lifecycle:
            preStop:
              exec:
                command:
                - /bin/kill
                - "-15"
                - "1"
          livenessProbe:
            httpGet:
              path: /_cluster/health?local=true
              port: http
            initialDelaySeconds: 300
          name: elasticsearch
          ports:
          - containerPort: 9200
            name: http
            protocol: TCP
          - containerPort: 9300
            name: transport
            protocol: TCP
          readinessProbe:
            httpGet:
              path: /_cluster/health
              port: http
          resources:
            limits:
              memory: 2Gi
          securityContext:
            capabilities:
              add:
              - IPC_LOCK
              - SYS_RESOURCE
          volumeMounts:
          - mountPath: /usr/share/elasticsearch/config/elasticsearch_custom.yml
            name: settings
            readOnly: true
            subPath: elasticsearch_custom.yml
        initContainers: null
        securityContext:
          fsGroup: 1000
        serviceAccountName: RELEASE-NAME-elasticsearch
        volumes:
        - configMap:
            items:
            - key: es.settings
              path: elasticsearch_custom.yml
            name: RELEASE-NAME-elasticsearch
          name: settings
